### DISCUSSION TITLE
What an awesome extension! Can we train it for personal use?

### DISCUSSION THREAD

**Discussion by le-khang** [Bing-su/adetailer#187] [Ideas]
Thanks so much for this awesome extension! It works great and is easy to use. Yesterday, I posted on Reddit about Teachable Machine and stumbled upon your extension https://www.reddit.com/r/StableDiffusion/comments/14q4srm/train_sd_for_caption_writing_im_tired_of/

Do you think it'll be possible in the next 1-3 years to train our own AI to recognize what we want (to write caption, to fix or rate things), like how we train LORA to draw things?

**Comment by SepticSpoons** [comment]
> Do you think it'll be possible in the next 1-3 years to train our own AI to recognize what we want (to write caption, to fix or rate things), like how we train LORA to draw things?

This is already possible. You use yolov8 ultralytics. There is docs over [here](https://docs.ultralytics.com/) if you want to read more about it. You just pip install ultralytics into a python venv, setup and label the dataset and train. Then you place your .pt file into models > adetailer folder to use it. I was thinking about making a more in-depth step by step guide on civitai detailing how to do it as it's easy once you know, but sounds more confusing than it is.

**Reply by Toad8787** [reply]
Thank you for making a tutorial:
https://civitai.com/articles/1224/training-a-custom-adetailer-model
